soroban-sdk/src/accounts.rs

```diff
@@ -1,550 +0,0 @@
-    }
-}
-
-impl AsRef<AccountId> for Account {
-    fn as_ref(&self) -> &AccountId {
-        &self.0
-    }
-}
-
-impl Account {
-    pub(crate) fn env(&self) -> &Env {
-        self.0.env()
-    }
-
-    pub(crate) fn as_raw(&self) -> &RawVal {
-        self.0.as_raw()
-    }
-
-    pub(crate) fn as_object(&self) -> &Object {
-        self.0.as_object()
-    }
-
-    pub(crate) fn to_raw(&self) -> RawVal {
-        self.0.to_raw()
-    }
-
-    pub(crate) fn to_object(&self) -> Object {
-        self.0.to_object()
-    }
-
-    pub fn to_id(&self) -> AccountId {
-        self.0.clone()
-    }
-
-    /// Returns if the account exists.
-    pub fn exists(id: &AccountId) -> bool {
-        let env = id.env();
-        env.account_exists(id.to_object())
-            .unwrap_infallible()
-            .is_true()
-    }
-
-    /// Returns the low threshold for the Stellar account.
-    pub fn low_threshold(&self) -> u8 {
-        let env = self.env();
-        let val = env
-            .account_get_low_threshold(self.to_object())
-            .unwrap_infallible();
-        let threshold_u32 = unsafe { <u32 as RawValConvertible>::unchecked_from_val(val) };
-        threshold_u32 as u8
-    }
-
-    /// Returns the medium threshold for the Stellar account.
-    pub fn medium_threshold(&self) -> u8 {
-        let env = self.env();
-        let val = env
-            .account_get_medium_threshold(self.to_object())
-            .unwrap_infallible();
-        let threshold_u32 = unsafe { <u32 as RawValConvertible>::unchecked_from_val(val) };
-        threshold_u32 as u8
-    }
-
-    /// Returns the high threshold for the Stellar account.
-    pub fn high_threshold(&self) -> u8 {
-        let env = self.env();
-        let val = env
-            .account_get_high_threshold(self.to_object())
-            .unwrap_infallible();
-        let threshold_u32 = unsafe { <u32 as RawValConvertible>::unchecked_from_val(val) };
-        threshold_u32 as u8
-    }
-
-    /// Returns the signer weight for the signer for this Stellar account. If
-    /// the signer does not exist for the account, returns zero (`0`).
-    pub fn signer_weight(&self, signer: &BytesN<32>) -> u8 {
-        let env = self.env();
-        let val = env
-            .account_get_signer_weight(self.to_object(), signer.to_object())
-            .unwrap_infallible();
-        let weight_u32 = unsafe { <u32 as RawValConvertible>::unchecked_from_val(val) };
-        weight_u32 as u8
-    }
-}
-
-#[cfg(any(test, feature = "testutils"))]
-use crate::testutils::{self, AccountId as _};
-
-#[cfg(any(test, feature = "testutils"))]
-#[cfg_attr(feature = "docs", doc(cfg(feature = "testutils")))]
-impl testutils::Accounts for Accounts {
-    fn generate_and_create(&self) -> AccountId {
-        let id = self.generate();
-        self.create(&id);
-        id
-    }
-
-    fn generate(&self) -> AccountId {
-        AccountId::random(self.env())
-    }
-
-    fn create(&self, id: &AccountId) {
```

Comment:
How do folks write tests with accounts without this? How do I create an account, with a particular balance, and use that balance in a test? Is there a new way to do this?

Reply:
The real question here is what would be the reason for creating an account with balance. So far I don't see one - you can just create a token with a given admin and use it to mint the balance (actually I think we'll be able to get rid even of the admin; thanks to the recording auth we can just mint in tests without providing any credentials, see https://github.com/stellar/rs-soroban-env/issues/653 for details). I think this is a much more natural UX, given that it doesn't go beyond what contracts already do.
As I've already mentioned, for the snapshot-based tests I would *hope* that the accounts/trustlines would be set up as a part of the snapshot itself. But if not, we can bring this back (maybe with a bit different API).

Reply:
Ok, lets see how far we can get without it, but we've been here before where we left it out, then later added it, etc. If we don't need it though, it will make for less code to maintain.

Reply:
I want to believe that the testing without accounts is much more plausible now (I think getting rid of invoker goes a long way in enabling this). But let's see indeed.